lightly/data/_helpers.py

```diff
@@ -34,104 +34,102 @@
+    """Checks whether the directory or any subdirectory contains video files.
 
     Args:
         root: Root directory path.
+        extensions: Tuple of valid video file extensions.
 
     Returns:
-        True if "root" or any subdir contains video files.
-
+        True if the root directory or any subdirectory contains video files, False otherwise.
     """
     for subdir, _, _ in os.walk(root):
         if _dir_contains_videos(subdir, extensions):
             return True
     return False
 
 
-def _is_lightly_output_dir(dirname: str):
+def _is_lightly_output_dir(dirname: str) -> bool:
     """Checks whether the directory is a lightly_output directory.
 
     Args:
-        dirname: Directory to check.
+        dirname: Directory name to check.
 
     Returns:
-        True if dirname is "lightly_outputs" else false.
-
+        True if the directory name is "lightly_outputs", False otherwise.
     """
     return "lightly_outputs" in dirname
 
 
-def _contains_subdirs(root: str):
-    """Checks whether directory contains subdirectories.
+def _contains_subdirs(root: str) -> bool:
+    """Checks whether the directory contains subdirectories.
 
     Args:
         root: Root directory path.
 
     Returns:
-        True if root contains subdirectories else false.
-
+        True if the root directory contains subdirectories (excluding "lightly_outputs"), False otherwise.
     """
     with os.scandir(root) as scan_dir:
         return any(not _is_lightly_output_dir(f.name) for f in scan_dir if f.is_dir())
 
 
 def _load_dataset_from_folder(
     root: str,
-    transform,
+    transform: Callable,
     is_valid_file: Optional[Callable[[str], bool]] = None,
-    tqdm_args: Dict[str, Any] = None,
+    tqdm_args: Optional[Dict[str, Any]] = None,
     num_workers_video_frame_counting: int = 0,
-):
-    """Initializes dataset from folder.
+) -> datasets.VisionDataset:
+    """Initializes a dataset from a folder.
+
+    This function determines the appropriate dataset type based on the contents of the root directory
+    and returns the corresponding dataset object.
 
     Args:
-        root: (str) Root directory path
-        transform: (torchvision.transforms.Compose) image transformations
+        root: Root directory path.
+        transform: Composed image transformations to be applied to the dataset.
+        is_valid_file: Optional function to determine valid files.
+        tqdm_args: Optional dictionary of arguments for tqdm progress bar.
+        num_workers_video_frame_counting: Number of workers for video frame counting.
 
     Returns:
-        Dataset consisting of images/videos in the root directory.
+        A dataset object (VideoDataset, ImageFolder, or DatasetFolder) based on the directory contents.
 
     Raises:
-        ValueError: If the specified dataset doesn't exist
-
+        ValueError: If the specified dataset directory doesn't exist or if videos are present
+                    but VideoDataset is not available.
```

Comment:
```suggestion
        ValueError: If the specified dataset directory doesn't exist or if videos are present
            but VideoDataset is not available.
```

We usually don't align indents but instead just indent one level more.